google/cloud/storage/internal/curl_client.cc

```diff
@@ -124,14 +127,14 @@ std::string HostHeader(Options const& options, char const* service) {
 Status CurlClient::SetupBuilderCommon(CurlRequestBuilder& builder,
                                       char const* method, char const* service) {
   auto auth_header =
-      opts_.get<Oauth2CredentialsOption>()->AuthorizationHeader();
+      CurrentOptions().get<Oauth2CredentialsOption>()->AuthorizationHeader();
   if (!auth_header.ok()) {
     return std::move(auth_header).status();
   }
   builder.SetMethod(method)
-      .ApplyClientOptions(opts_)
+      .ApplyClientOptions(CurrentOptions())
       .AddHeader(auth_header.value())
-      .AddHeader(HostHeader(opts_, service))
+      .AddHeader(HostHeader(CurrentOptions(), service))
```

Comment:
It is probably worth doing `auto const& options = CurrentOptions();` once as finding the options is like doing a map lookup by tid.

Reply:
> It is probably worth doing `auto const& options = CurrentOptions();` once

Done.

> as finding the options is like doing a map lookup by tid.

That would be a really sad implementation of thread local storage.  My recollection is that there is a register with the thread context, and some googling seems to confirm that:

https://chao-tic.github.io/blog/2018/12/25/tls

Once your find the control block for the thread, all the TLS variables are in per-thread array, and should be an indirection (or two or three) into that register.

Reply:
Yes ... right you are.  I'm a Neanderthal.  Certainly lessens the need to factor `CurrentOptions()`, but still not a bad thing.